dataikuapi/dss/sqlnotebook.py

```diff
@@ -0,0 +1,206 @@
+                                               (self.project_key, self.notebook_id))
+        return DSSNotebookHistory(self.client, self.project_key, self.notebook_id, query_runs)
+
+    def delete(self):
+        """
+        Delete this SQL notebook
+        """
+        return self.client._perform_json("DELETE",
+                                         "/projects/%s/sql-notebooks/%s" % (self.project_key, self.notebook_id))
+
+    ########################################################
+    # Discussions
+    ########################################################
+    def get_object_discussions(self):
+        """
+        Get a handle to manage discussions on the notebook
+
+        :return: The handle to manage discussions
+        :rtype: :class:`dataikuapi.dss.discussion.DSSObjectDiscussions`
+        """
+        return DSSObjectDiscussions(self.client, self.project_key, "SQL_NOTEBOOK", self.notebook_id)
+
+class DSSNotebookContent(object):
+    """
+    The content of a SQL notebook
+
+    .. important::
+
+        Do not instantiate directly, use :meth:`dataikuapi.dss.sqlnotebook.DSSSQLNotebook.get_content`
+    """
+    def __init__(self, client, project_key, notebook_id, content):
+        self.client = client
+        self.project_key = project_key
+        self.notebook_id = notebook_id
+        self.content = content
+
+    def get_raw(self):
+        """
+        Get the raw content of this SQL notebook
+
+        The content comprises cells
+
+        :return: A dict containing the full content of a notebook. Notable fields are:
+
+            * **cells**: list of cells, each one a dict (see :meth:`get_cells()`)
+
+        :rtype: dict
+        """
+        return self.content
+
+    def get_cells(self):
+        """
+        Get the cells of this SQL notebook
+
+        :return: A list of cells. Each cell is a dict, with notable fields:
+
+            * **type**: type of the cell, for example 'QUERY' or 'MARKDOWN'
+            * **name**: name of the cell
+            * **code**: content of the cell
+
+        :rtype: list[dict]
+        """
+        return self.content["cells"]
+
+    def save(self):
+        """
+        Save the content of this SQL notebook
+        """
+        return self.client._perform_json("PUT",
+                                         "/projects/%s/sql-notebooks/%s" % (self.project_key, self.notebook_id),
+                                         body=self.content)
+
+class DSSNotebookHistory(object):
+    """
+    The history of a SQL notebook
+
+    .. important::
+
+        Do not instantiate directly, use :meth:`dataikuapi.dss.sqlnotebook.DSSSQLNotebook.get_history`
+    """
+    def __init__(self, client, project_key, notebook_id, query_runs):
+        self.client = client
+        self.project_key = project_key
+        self.notebook_id = notebook_id
+        self.query_runs = query_runs
+
+    def get_query_runs(self):
```

Comment:
I'm deliberately calling these query runs instead of queries. LMK if that's alright or we should use a different name.